### PR TITLE
Rehan/envoy subdomain

### PR DESCRIPTION
**Pull request by RehanSD**
Must merge other PRs in. Put so progress can be seen

### CODE REVIEW

File: go/mock_server.go [hunk 1]
```diff
@@ -256,8 +258,14 @@ func main() {
 		panic(err)
 	}
 	log.Println("Server started, listening to port", port)
-	// Change this to whatever database you want to use.
-	db, err = gorm.Open("postgres", "host=34.213.216.228 port=5432 user=modelzoo")
+	log.Println("Running in", os.Args[1], "mode.")
+	if os.Args[1] == "public" {
```
Comment (simon-mo) [inline go/mock_server.go]: This is find for now, but in future we should try to use some golang arg parsing libraries.

File: go/mock_server.go [hunk 2]
```diff
@@ -256,8 +258,14 @@ func main() {
...
+		db, err = gorm.Open("postgres", "host=34.213.216.228 port=5432 user=modelzoo")
```
Comment (simon-mo) [inline go/mock_server.go]: Add a line TODO marking this should not be hardcoded.

File: go/setup_db.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	db.Create(&dbtypes.User{ID: iuid, UserName: "admin", Email: "admin@admin.com", Token: "admin_token", LastUsed: time.Now()})
+
+	// Define base models
+	var models = [...]string{"res50-pytorch", "squeezenet-pytorch", "rise-pytorch", "marvel-pytorch", "image-segmentation", "image-captioning"}
+	for _, mod := range models {
+		cat, out := cats(mod)
+		db.Create(&dbtypes.Model{ID: uuid.New(), Name: mod, Author: iuid, ModelCategory: cat, OutputType: out, Private: false})
+	}
+}
+func populate_cats(db *gorm.DB) {
```
Comment (simon-mo) [inline go/setup_db.go]: Can these be ran in a for-loop?

File: go/setup_db.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	db.Exec("INSERT INTO categories VALUES (1,  \"TextGeneration\")")
+	db.Exec("INSERT INTO categories VALUES (2, \"ImageSegmentation\")")
+	db.Exec("INSERT INTO categories VALUES (3,\"ImageCaptioning\")")
+}
+func populate_outputs(db *gorm.DB) {
+	db.Exec("INSERT INTO output_types VALUES (0, \"vision\")")
+	db.Exec("INSERT INTO output_types VALUES (1, \"text\")")
+	db.Exec("INSERT INTO output_types VALUES (2, \"segment\")")
+}
+func cats(m string) (int, int) {
```
Comment (simon-mo) [inline go/setup_db.go]: 1. expand `cats`
2. don't use string literal, use CONSTANTS

**Comment by simon-mo** [comment]
One more round to go. Great work!

**Comment by simon-mo** [comment]
Quick ping on this. And fixing the python package comments in closed PR

File: go/setup_db.go (hunk 4 above)
Comment (RehanSD) [reply go/setup_db.go]: what do you mean by constants? Like declare them as const RES50_PYTORCH = "res50-pytorch"?

File: models.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+    "models": [
+        {
+            "name":"Rehan",
```
Comment (simon-mo) [inline models.json]: 🤔

File: models.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+            "category":"VISIONCLASSIFICATION",
+            "output":"ViSIon"
+        },
+        {
+            "name":"aehan",
```
Comment (simon-mo) [inline models.json]: we are really using these models?

**Pull request merged**